Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h

```diff
@@ -60,6 +60,21 @@ namespace AZ
         class ObjectStreamImpl;
     }
 
+    namespace SerializeContextAttributes
```

Comment:
This definition was moved from ObjectStream in order to prevent compile and runtime errors returning the wrong type info on Clang builds.

Reply:
Yes, this was causing problems with Clang compilers due to the VariantReflection.inl file not being able to see the `AZ_TYPE_INFO_SPECIALIZE` call for ObjectStreamWriteOverrideResponse enum and therefore returnning a NullUuid instead of `"{BDF960A8-0F18-4E9D-96DA-F800A122C42D}"` when aggregating the TypeId for the [ObjectStreamWriter](https://github.com/o3de/o3de/pull/8623/files#diff-8dbcd063d898d36e8d0e4989325341a3b60aa9828a9eab03fe170e3215b89f43R436) function pointer.